packages/starlight/utils/user-config.ts

```diff
@@ -35,6 +35,11 @@ const SidebarBaseSchema = z.object({
   translations: z.record(z.string()).default({}),
 });
 
+const SidebarCollapsibleItemSchema = z.object({
+  /** Whether this item should be collapsed by default. */
+  collapsed: z.boolean().default(false),
+});
```

Comment:
I think I might suggest extending here already, something like this:

```suggestion
const SidebarGroupSchema = SidebarBaseSchema.extend({
  /** Whether this item should be collapsed by default. */
  collapsed: z.boolean().default(false),
});
```

And then using `SidebarGroupSchema` _instead_ of the base schema for the individual group types. Would avoid some of the awkwardness of the `&` intersections and the `merge` call for `ManualSidebarGroupSchema`.

Reply:
Way better indeed 👍🏼